friends/FriendsHandler.js

```diff
@@ -0,0 +1,243 @@
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+    return { 
+      statusCode: 403,
+      body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+    };
+  
+  return await getRequests(decoded.id)
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/*
+ * POST /friends/request/send/
+ */
+module.exports.sendRequest = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  const token = event.headers.Authorization;
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+    return { 
+      statusCode: 403,
+      body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+    };
+
+  return await sendRequest(decoded.id, JSON.parse(event.body))
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/*
+ * POST /friends/request/accept/
+ */
+module.exports.acceptRequest = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  const token = event.headers.Authorization;
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+    return { 
+      statusCode: 403,
+      body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+    };
+
+  return await acceptRequest(decoded.id, JSON.parse(event.body))
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/*
+ * POST /friends/request/decline/
+ */
+module.exports.declineRequest = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  const token = event.headers.Authorization;
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+  return { 
+    statusCode: 403,
+    body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+  };
+
+  return await declineRequest(decoded.id, JSON.parse(event.body))
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/**
+ * Helpers
+ */
+
+function decodeJwt(token) {
+  console.log("decodeJwt");
+
+  try {
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
+    return decoded;
+  } catch(err) {
+    console.error("decodeJwt error " + err);
+    return null;
+  }
+}
+
+function getFriends(id) {
+  console.log("getFriends");
```

Comment:
I don't think that we need to be logging every function call like this. It'll clutter up the logs and we can get stack traces from errors anyway.